### PR TITLE
Convert ReactFresh to createRoot

### PR DESCRIPTION
**Pull request by eps1lon**

### CODE REVIEW

File: packages/react-refresh/src/__tests__/ReactFresh-test.js [hunk 1]
```diff
@@ -2448,16 +2464,15 @@ describe('ReactFresh', () => {
...
     await waitFor(['App#layout']);
     const el = container.firstChild;
     expect(el.hidden).toBe(true);
     expect(el.firstChild).toBe(null); // Offscreen content not flushed yet.
 
     // Perform a hot update.
-    patch(() => {
-      function Hello() {
+    {
+      const Hello = () => {
```
Comment (eps1lon) [inline packages/react-refresh/src/__tests__/ReactFresh-test.js:2475]: When I change this to `await patch` i.e. wrapping `performReactRefresh` in `act` I get considerably different assertions. Seems like this test wants to assert on some intermediate states but I don't know how to port them

File: packages/react-refresh/src/__tests__/ReactFresh-test.js [hunk 2]
```diff
@@ -1305,7 +1313,7 @@ describe('ReactFresh', () => {
 
   it('can patch both trees while suspense is displaying the fallback', async () => {
```
Comment (eps1lon) [inline packages/react-refresh/src/__tests__/ReactFresh-test.js:1314]: This test is no longer passing. Seems like it really can't patch both trees anymore. @gaearon Could you take a look if there's a legit bug or something we just can't test in new roots anymore.

File: packages/react-refresh/src/__tests__/ReactFresh-test.js [hunk 3]
```diff
@@ -1411,19 +1416,19 @@ describe('ReactFresh', () => {
...
       });
 
-      // Colors inside both trees should change:
+      // Only update color in the visible child
       expect(container.childNodes.length).toBe(2);
       expect(container.childNodes[0]).toBe(primaryChild);
       expect(container.childNodes[1]).toBe(fallbackChild);
       expect(primaryChild.textContent).toBe('Content 1');
-      expect(primaryChild.style.color).toBe('red');
+      expect(primaryChild.style.color).toBe('green');
```
Comment (eps1lon) [inline packages/react-refresh/src/__tests__/ReactFresh-test.js:1424]: This test is no longer passing. Seems like it really can't patch both trees anymore. @gaearon Could you take a look if there's a legit bug or something we just can't test in new roots anymore.

File: packages/react-refresh/src/__tests__/ReactFresh-test.js [hunk 4]
```diff
@@ -1910,26 +1915,31 @@ describe('ReactFresh', () => {
...
-          runRemountingStressTest(trees[j]);
-          runRemountingStressTest(trees[i]);
+          await runRemountingStressTest(trees[i]);
+          await runRemountingStressTest(trees[j]);
+          await runRemountingStressTest(trees[i]);
         }
       }
     }
-  });
+  }, 10000);
```
Comment (eps1lon) [inline packages/react-refresh/src/__tests__/ReactFresh-test.js:1940]: This test now consistently times out in CI: https://app.circleci.com/pipelines/github/facebook/react/49942/workflows/4d2a2a4a-54ee-4ed3-9e1d-80c7bb4b0e28/jobs/767976

File: packages/react-refresh/src/__tests__/ReactFresh-test.js (hunk 3 above)
Comment (gaearon) [reply packages/react-refresh/src/__tests__/ReactFresh-test.js:1424]: I'm not sure but I suspect maybe we don't bother re-rendering into the invisible component because it's offscreen? I'd rephrase the test so that it doesn't assert that it *has to* be green or red here — I think the important part is that by the time we unsuspend, the color of the content is up-to-date. Until then, I think it doesn't really matter because it's unobservable.

**Review by gaearon — APPROVED** [review]

File: packages/react-refresh/src/__tests__/ReactFresh-test.js (hunk 3 above)
Comment (eps1lon) [reply packages/react-refresh/src/__tests__/ReactFresh-test.js:1424]: Test now reads "only patches the fallback tree while suspended"
